src/pydantic_zarr/v2.py

```diff
@@ -823,9 +820,12 @@ def to_zarr(
     return spec.to_zarr(store, path, overwrite=overwrite, **kwargs)
 
 
-def to_flat(node: ArraySpec | GroupSpec, root_path: str = "") -> dict[str, ArraySpec | GroupSpec]:
+def to_flat(
+    node: ArraySpec[Any] | GroupSpec[Any, Any], root_path: str = ""
```

Comment:
Is there a reason these can't be `AnyArraySpec` and `AnyGroupSpec`?

Reply:
i don't think so, i'll make that change